api/src/utils/transformations.ts

```diff
@@ -63,3 +98,71 @@ export function maybeExtractFormat(transforms: Transformation[]): string | undef
 	const lastToFormat = toFormats[toFormats.length - 1];
 	return lastToFormat ? lastToFormat[1]?.toString() : undefined;
 }
+
+type Dimensions = { w: number; h: number };
+
+/**
+ * Resize an image but keep it centered on the focal point.
+ * Based on the method outlined in https://github.com/lovell/sharp/issues/1198#issuecomment-384591756
+ */
+function getResizeArguments(
+	original: Dimensions,
+	target: Dimensions,
+	focalPoint?: FocalPoint | null,
+): { width: number; height: number; region: Region } {
+	const { width, height, factor } = getIntermediateDimensions(original, target);
+
+	const region = getExtractionRegion(factor, focalPoint ?? { x: original.w / 2, y: original.h / 2 }, target, {
+		w: width,
+		h: height,
+	});
+
+	return { width, height, region };
+}
+
+/**
+ * Calculates the dimensions of the intermediate (resized) image.
+ */
+function getIntermediateDimensions(
+	original: Dimensions,
+	target: Dimensions,
+): { width: number; height: number; factor: number } {
+	const hRatio = original.h / target.h;
+	const wRatio = original.w / target.w;
+
+	let factor: number;
+	let width: number;
+	let height: number;
+
+	if (hRatio < wRatio) {
+		factor = hRatio;
+		height = Math.round(target.h);
+		width = Math.round(original.w / factor);
+	} else {
+		factor = wRatio;
+		width = Math.round(target.w);
+		height = Math.round(original.h / factor);
+	}
```

Comment:
Will this work as expected for images where `hRatio == wRatio` ?

Reply:
Yes. Here are some debugging outputs for width:

```
api dev: --- 100x100 -> 98x100: { width: 100, height: 100, factor: 1 }
api dev: --- 100x100 -> 99x100: { width: 100, height: 100, factor: 1 }
api dev: --- 100x100 -> 100x100: { width: 100, height: 100, factor: 1 }
api dev: --- 100x100 -> 101x100: { width: 101, height: 101, factor: 0.9900990099009901 }
api dev: --- 100x100 -> 102x100: { width: 102, height: 102, factor: 0.9803921568627451 }
```

And for height:

```
api dev: --- 100x100 -> 100x98: { width: 100, height: 100, factor: 1 }
api dev: --- 100x100 -> 100x99: { width: 100, height: 100, factor: 1 }
api dev: --- 100x100 -> 100x100: { width: 100, height: 100, factor: 1 }
api dev: --- 100x100 -> 100x101: { width: 101, height: 101, factor: 0.9900990099009901 }
api dev: --- 100x100 -> 100x102: { width: 102, height: 102, factor: 0.9803921568627451 }
```

We can see that

1. If the target-dimensions are smaller than our original, the factor correctly doesnt change, since the target can be fully contained
2. If the target is equal to our original we also dont need to scale, factor stays
3. But if the target is larger, then we have to set the factor in order to scale the original so that we can fully cover the target